sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -126,10 +126,9 @@ def _build_create_table_exp(
             # this checks the explain plain from redshift and tries to detect when these optimizer
             # bugs occur and force a cast
             explain_statement = statement.copy()
-            for select in explain_statement.find_all(exp.Select):
-                if select.args.get("from"):
-                    select.set("limit", None)
-                    select.set("where", None)
+            for select_or_union in explain_statement.find_all(exp.Select, exp.Union):
+                select_or_union.set("limit", None)
+                select_or_union.set("where", None)
```

Comment:
is there any concern of a potential cross join here? in sql you can do 

select 
from x, y
where xid =yid

removing the where significantly alters the query but maybe won’t matter

Reply:
select_or_union.pop("limit")
select_or_union.pop("where")